core-bundle/src/Command/DeferredImagesCommand.php

```diff
@@ -0,0 +1,243 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Command;
+
+use Contao\CoreBundle\Image\ImageFactoryInterface;
+use Contao\Image\DeferredImageInterface;
+use Contao\Image\DeferredImageStorageInterface;
+use Contao\Image\DeferredResizerInterface;
+use Contao\Image\ResizerInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Exception\InvalidArgumentException;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Terminal;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Process\PhpExecutableFinder;
+use Symfony\Component\Process\Process;
+
+/**
+ * Resize deferred images that have not been processed yet.
+ */
+class DeferredImagesCommand extends Command
+{
+    /**
+     * @var Filesystem
+     */
+    private $filesystem;
+
+    /**
+     * @var ImageFactoryInterface
+     */
+    private $imageFactory;
+
+    /**
+     * @var ResizerInterface
+     */
+    private $resizer;
+
+    /**
+     * @var string
+     */
+    private $targetDir;
+
+    /**
+     * @var DeferredImageStorageInterface
+     */
+    private $storage;
+
+    /**
+     * @var int
+     */
+    private $terminalWidth;
+
+    public function __construct(Filesystem $filesystem, ImageFactoryInterface $imageFactory, ResizerInterface $resizer, string $targetDir, DeferredImageStorageInterface $storage)
+    {
+        $this->filesystem = $filesystem;
+        $this->imageFactory = $imageFactory;
+        $this->resizer = $resizer;
+        $this->targetDir = $targetDir;
+        $this->storage = $storage;
+        $this->terminalWidth = (new Terminal())->getWidth();
+
+        parent::__construct();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure(): void
+    {
+        $this
+            ->setName('contao:deferred-images')
+            ->setDefinition([
+                new InputOption('time-limit', 'l', InputArgument::OPTIONAL, 'Time limit in seconds', '0'),
+                new InputOption('concurrent', 'c', InputArgument::OPTIONAL, 'Run multiple processes concurrently', '1'),
+                new InputOption('throttle', 't', InputArgument::OPTIONAL, 'Pause between resizes to limit cpu utilization, 0.1 relates to 10% cpu usage', '1'),
```

Comment:
Sounds a bit complicated to me at first sight. Wouldn't a simple `delay` be enough? :)

Reply:
The time how long a single resize takes differs very much, so a fixed delay might not be practicable if I want to limit cpu usage.

Reply:
Yeah I think it's more that I don't understand why it's there because I'm not familiar enough with the code :)

Reply:
In some shared hosting environments, having a long running process using 100% cpu is undesirable. With the throttle parameter you can define how high the average cpu usage should be.